### PR TITLE
chore: minor re-organization of integration tests and utils

### PR DESCRIPTION
**Pull request by cprice404**
This commit just attempts to re-organize the int test file hierarchy a bit to make it easier to navigate in the future. Of note:

* Make a subdir for 'cache' integration tests so we can easily separate topic and other int tests in the future
* Move the SimpleCacheClient int tests into an `old` subdir to make it obvious they are going away
* Move the `cache_test_state` stuff up into the existing `test-util` dir, where the doctest helpers are defined, to consolidate all the utility/helper code into the same spot.

### CODE REVIEW

**Review by nand4011 — APPROVED** [review]

**Review by pratik151192 — APPROVED** [review]

File: test-util/Cargo.toml [hunk 1]
```diff
@@ -10,3 +10,4 @@ anyhow = "1.0.68"
 tokio = { version = "1.25.0", features = ["full"] }
 uuid = { version = "1.2.2", features = ["v4"] }
 scopeguard = "1.1.0"
+once_cell = "1.19.0"
```
Comment (malandis) [inline test-util/Cargo.toml:13]: Nice catch

File: test-util/src/test_utils.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+            // If any background tasks were spawned we give them some time to exit cleanly.
+            runtime.shutdown_timeout(Duration::from_secs(1));
+        }
+    });
+
+    runtime.block_on(func(cache_name, credential_provider))
+}
+
+pub fn create_doctest_client() -> (CacheClient, String) {
```
Comment (malandis) [inline test-util/src/test_utils.rs:52]: Rename to `create_doctest_cache_client`?

**Review by malandis — COMMENTED** [review]
One suggestion. Looks good otherwise 👍

File: test-util/src/test_utils.rs (hunk 2 above)
Comment (cprice404) [reply test-util/src/test_utils.rs]: fixed in https://github.com/momentohq/client-sdk-rust/pull/193/commits/ae6f8e330f70d2990eee68d32bf8adb2f1760c0e

**Review by pgautier404 — APPROVED** [review]

**Pull request merged**